File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {
        int small = arrays[0][0], big = arrays[0].back();
        int result = 0;

        for (int i = 1; i < arrays.size(); i++) {
            result = max(result, abs(arrays[i].back() - small));
            result = max(result, abs(big - arrays[i][0]));
            small = min(small, arrays[i][0]);
            big = max(big, arrays[i].back());
        }

        return result;
    }
};

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.py
class Solution:
    def maxDistance(self, arrays: List[List[int]]) -> int:
        small, big, result = arrays[0][0], arrays[0][-1], 0

        for array in arrays[1:]:
            result = max(result, abs(array[-1] - small), abs(big - array[0]))
            small = min(small, array[0])
            big = max(big, array[-1])

        return result